### PR TITLE
Implements the share UI

### PR DESCRIPTION
**Pull request by chuga**
This introduces the option to turn on or off the sharing when using Firebase.
This was implemented according to the mocks, but we'll probably revisit the ShareLinkCell soon.

### CODE REVIEW

File: src/core/UI/RMXOverlayViewController.m [hunk 1]
```diff
@@ -248,20 +274,57 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
...
+
+- (void)cellRequestedFullScreenOverlay:(RMXCell *)cell {
+  [self maximizePanel];
+}
 
-- (void)touchStartedAtPoint:(CGPoint)point withEvent:(UIEvent *)event {
-  // No-op.
+#pragma mark - RMXShareLinkCellDelegate
+
+- (void)linkButtonWasTapped:(UIButton *)linkButton {
```
Comment (chriscox) [inline src/core/UI/RMXOverlayViewController.m:285]: Are you still going to open the URL here? Or will this instead copy link?

Comment (chuga) [reply src/core/UI/RMXOverlayViewController.m:285]: The cell with the link needs to be revisited but this PR is already pretty long so I'd rather get it in and then make changes to it in a separate PR.

File: src/core/RMXRemixer.m [hunk 2]
```diff
@@ -202,4 +197,21 @@ + (void)updateVariable:(RMXVariable *)variable fromRemoteControllerToValue:(id)v
...
 }
 
+#pragma mark - Remote controller
+
++ (nullable NSURL *)remoteControllerURL {
+  return [[[self sharedInstance] remoteController] remoteControllerURL];
+}
+
++ (BOOL)isSharing {
+  if (![[self sharedInstance] remoteController]) {
```
Comment (chriscox) [inline src/core/RMXRemixer.m]: Can you skip this `if` clause and simply return `return [[[self sharedInstance] remoteController] isSharing];` ?

And if not, then won't `applicationDidBecomeActive` and `...resign` fail as well?

File: src/core/UI/RMXOverlayViewController.m (hunk 1 above)
Comment (chriscox) [reply src/core/UI/RMXOverlayViewController.m:285]: OK sounds good.

**Review by chriscox — APPROVED** [review]
Minor comment. Otherwise LGTM.

File: src/core/RMXRemixer.m (hunk 2 above)
Comment (chuga) [reply src/core/RMXRemixer.m]: you're right, I didn't need it. Fixed.

**Pull request merged**